File: java/javadpbo/src/com/sem4dpbo/week6/contohpolymorfism/Buku.java
package com.sem4dpbo.week6.contohpolymorfism;

public class Buku {
    private String judul;
    private int tahun;
    private String pengarang;

    Buku(){}

    Buku(String judul, int tahun, String pengarang){
        this.judul = judul;
        this.tahun = tahun;
        this.pengarang = pengarang;
    }

    public void setBuku(String judul){
        this.judul = judul;
    }

    public void setBuku(String judul, int tahun){
        this.judul = judul;
        this.tahun = tahun;
    }

    public void setBuku(String judul, int tahun, String pengarang){
        this.judul = judul;
        this.tahun = tahun;
        this.pengarang = pengarang;
    }

    public String getJudul() {
        return judul;
    }

    public void setJudul(String judul) {
        this.judul = judul;
    }

    public int getTahun() {
        return tahun;
    }

    public void setTahun(int tahun) {
        this.tahun = tahun;
    }

    public String getPengarang() {
        return pengarang;
    }

    public void setPengarang(String pengarang) {
        this.pengarang = pengarang;
    }
}

File: cpp/6th_Week/polimorfisme/overloading/Buku.cpp
#include <iostream>
#include <string>

using namespace std;

class Buku
{
private:
    string judul;
    int tahun;
    string pengarang;
public:
    Buku() {}

    Buku(string judul, int tahun, string pengarang){
        this->judul = judul;
        this->tahun = tahun;
        this->pengarang = pengarang;
    }

    void setBuku(string judul){
        this->judul = judul;
    }

    void setBuku(string judul, int tahun){
        this->judul = judul;
        this->tahun = tahun;
    }

    void setBuku(string judul, int tahun, string pengarang){
        this->judul = judul;
        this->tahun = tahun;
        this->pengarang = pengarang;
    }

    void setJudul(string judul){
        this->judul = judul;
    }

    void setTahun(int tahun){
        this->tahun = tahun;
    }

    void setPengarang(string pengarang){
        this->pengarang = pengarang;
    }

    string getJudul(){
        return this->judul;
    }

    int getTahun(){
        return this->tahun;
    }

    string getPengarang(){
        return this->pengarang;
    }

    ~Buku() {}
};